cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -2075,7 +2075,7 @@ Status ExecRound(KernelContext* ctx, const ExecBatch& batch, Datum* out) {
 // kernel dispatch based on RoundMode, only on non-null output.
 template <template <typename, RoundMode, typename...> class Op, typename OptionsType>
 std::shared_ptr<ScalarFunction> MakeUnaryRoundFunction(std::string name,
-                                                       const FunctionDoc* doc) {
+                                                       FunctionDoc doc) {
   using State = RoundOptionsWrapper<OptionsType>;
   static const OptionsType kDefaultOptions = OptionsType::Defaults();
   auto func = std::make_shared<ArithmeticIntegerToFloatingPointFunction>(
```

Comment:
Below this line there is another spot for std::move(doc) but Github won't let me suggest it.

Reply:
I found another few places too, I updated them as well.